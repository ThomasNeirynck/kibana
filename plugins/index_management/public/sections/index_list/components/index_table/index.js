export { IndexTable } from './index_table.container';
