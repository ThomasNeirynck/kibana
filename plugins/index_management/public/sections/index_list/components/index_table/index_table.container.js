import { connect } from 'react-redux';
import {
  getDetailPanelIndexName,
  getPageOfIndices,
  getPager,
  getFilter,
  isDetailPanelOpen,
  showSystemIndices,
  getSortField,
  isSortAscending
} from '../../../../store/selectors';
import {
  filterChanged,
  closeDetailPanel,
  openDetailPanel,
  pageChanged,
  pageSizeChanged,
  sortChanged,
  showSystemIndicesChanged,
} from '../../../../store/actions';

import { IndexTable as PresentationComponent } from './index_table';

const mapStateToProps = (state) => {
  return {
    isDetailPanelOpen: isDetailPanelOpen(state),
    detailPanelIndexName: getDetailPanelIndexName(state),
    indices: getPageOfIndices(state),
    pager: getPager(state),
    filter: getFilter(state),
    showSystemIndices: showSystemIndices(state),
    sortField: getSortField(state),
    isSortAscending: isSortAscending(state)
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    filterChanged: (filter) => {
      dispatch(filterChanged({ filter }));
    },
    pageChanged: (pageNumber) => {
      dispatch(pageChanged({ pageNumber }));
    },
    pageSizeChanged: (pageSize) => {
      dispatch(pageSizeChanged({ pageSize }));
    },
    sortChanged: (sortField, isSortAscending) => {
      dispatch(sortChanged({ sortField, isSortAscending }));
    },
    showSystemIndicesChanged: (showSystemIndices) => {
      dispatch(showSystemIndicesChanged({ showSystemIndices }));
    },
    openDetailPanel: (indexName) => {
      dispatch(openDetailPanel({ indexName }));
    },
    closeDetailPanel: () => {
      dispatch(closeDetailPanel());
    }
  };
};

export const IndexTable = connect(
  mapStateToProps,
  mapDispatchToProps
)(PresentationComponent);
