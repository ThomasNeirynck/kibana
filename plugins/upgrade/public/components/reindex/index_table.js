import { sortBy } from 'lodash';
import classNames from 'classnames';
import React from 'react';

import { RefreshButton } from '../refresh_button';
import { KuiToolBar } from 'ui_framework/components';
import { IndexTableRow } from './index_table_row';


export function IndexTable({
  cancelAction,
  className,
  indices,
  loadIndices,
  processIndex,
  resetAction,
}) {
  const classes = classNames('kuiControlledTable', className);
  const sortedIndices = sortBy(indices, 'name');

  return (
    <div className={ classes }>
      <KuiToolBar>
        <div className="kuiToolBarSection">

          <RefreshButton
            buttonLabel="Refresh Indices"
            onClick={ loadIndices }
          />

        </div>
        <div className="kuiToolBarSection">
          <div className="kuiToolBarText">
            { sortedIndices.length } indices
          </div>
        </div>
      </KuiToolBar>
      <table className="kuiTable">
        <thead>
          <tr>
            <th className="kuiTableHeaderCell">
              Index Name
            </th>
            <th className="kuiTableHeaderCell">
              Required Action
            </th>
            <th className="kuiTableHeaderCell kuiTableHeaderCell--alignRight">
              Task Management
            </th>
          </tr>
        </thead>
        { sortedIndices.map((index) => (
          <IndexTableRow
            cancelAction={ cancelAction }
            index={ index }
            key={ index.name }
            resetAction={ resetAction }
            processIndex={ processIndex }
          />
        )) }
      </table>
    </div>
  );
}

IndexTable.propTypes = {
  cancelAction: React.PropTypes.func,
  className: React.PropTypes.string,
  indices: React.PropTypes.objectOf(React.PropTypes.object),
  loadIndices: React.PropTypes.func,
  resetAction: React.PropTypes.func,
  processIndex: React.PropTypes.func,
};

IndexTable.defaultProps = {
  cancelAction: () => {},
  className: '',
  indices: {},
  loadIndices: () => {},
  resetAction: () => {},
  processIndex: () => {},
};

