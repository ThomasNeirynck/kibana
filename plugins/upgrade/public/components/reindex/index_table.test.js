import _ from 'lodash';
import { shallow } from 'enzyme';
import React from 'react';

import { IndexTable } from './index_table';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function Notifier() {},
}), { virtual: true });


describe('IndexTable', () => {
  test('renders the index count', () => {
    const component = (
      <IndexTable
        indices={ {
          index2: { name: 'index1' },
          index1: { name: 'index2' },
          index3: { name: 'index3' },
        } }
      />
    );

    const wrapper = shallow(component);

    expect(wrapper.findWhere((node) => node.childAt(0).text() === '3 indices').exists()).toBe(true);
  });

  test('renders an IndexTableRow for every index sorted by name', () => {
    const component = (
      <IndexTable
        indices={ {
          index2: { name: 'index1' },
          index1: { name: 'index2' },
          index3: { name: 'index3' },
        } }
      />
    );

    const wrapper = shallow(component);

    expect(wrapper.find('IndexTableRow').map(_.identity)).toMatchSnapshot();
  });
});
