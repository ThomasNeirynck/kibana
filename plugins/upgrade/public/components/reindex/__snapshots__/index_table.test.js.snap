// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexTable renders an IndexTableRow for every index sorted by name 1`] = `
Array [
  <IndexTableRow
    cancelAction={[Function]}
    index={
      Object {
        "name": "index1",
      }
    }
    processIndex={[Function]}
    resetAction={[Function]}
  />,
  <IndexTableRow
    cancelAction={[Function]}
    index={
      Object {
        "name": "index2",
      }
    }
    processIndex={[Function]}
    resetAction={[Function]}
  />,
  <IndexTableRow
    cancelAction={[Function]}
    index={
      Object {
        "name": "index3",
      }
    }
    processIndex={[Function]}
    resetAction={[Function]}
  />,
]
`;
